File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.js
/**
 * @param {string} text
 * @return {number}
 */
var maxNumberOfBalloons = function (text) {
    let map = new Map()
    for (let i = 0; i < text.length; i++) {
        map.set(text[i], (map.get(text[i]) || 0) + 1)
    }
    let min = Infinity
    let word = "balloon"
    if (map.size < 5) {
        return 0
    }
    map.set("l", Math.floor(map.get("l") / 2))
    map.set("o", Math.floor(map.get("o") / 2))
    for (let i = 0; i < word.length; i++) {
        min = Math.min(min, map.get(word[i]) || 0)
    }
    return min
};

File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        map = defaultdict(int)
        for i in range(len(text)):
            map[text[i]] += 1

        min_count = float('inf')
        word = "balloon"
        if len(map) < 5:
            result = 0
        else:
            map["l"] = math.floor(map["l"] / 2)
            map["o"] = math.floor(map["o"] / 2)
            for i in range(len(word)):
                min_count = min(min_count, map[word[i]])
            result = min_count

        return result

